src/codegen/llvm/codegen_llvm_helper_visitor.cpp

```diff
@@ -251,7 +251,8 @@ static void append_statements_from_block(ast::StatementVector& statements,
     for (const auto& statement: block_statements) {
         const auto& expression_statement = std::dynamic_pointer_cast<ast::ExpressionStatement>(
             statement);
-        if (!expression_statement->get_expression()->is_solve_block())
+        if (!expression_statement ||
+            expression_statement && !expression_statement->get_expression()->is_solve_block())
```

Comment:
Isn't `expression_statement` in  `expression_statement && ....` redundant? (because we have `!expression_statement ||`)

Reply:
It is! Should be `!expression_statement || expression_statement->get_expression()->is_solve_block()`